modules/matterhorn-serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/ServiceRegistryJpaImpl.java

```diff
@@ -2502,12 +2502,14 @@ else if (job.getStatus() == Status.FINISHED) {
       // Services in WARNING state triggered by current job
       List<ServiceRegistrationJpaImpl> relatedWarningServices = getRelatedWarningServices(job);
 
-      // Sets all related services to error state
+      // The related services are already in WARNING state and max attempts is reached
       for (ServiceRegistrationJpaImpl relatedService : relatedWarningServices) {
-        logger.info("State set to ERROR for related service {} on host {}", currentService.getServiceType(),
-                currentService.getHost());
-        relatedService.setServiceState(ERROR, job.toJob().getSignature());
-        updateServiceState(em, relatedService);
+        if (!currentService.equals(relatedService) && getHistorySize(relatedService) >= maxAttemptsBeforeErrorState) {
+          logger.info("State set to ERROR for related service {} on host {}", relatedService.getServiceType(),
+                  relatedService.getHost());
+          relatedService.setServiceState(ERROR, job.toJob().getSignature());
+          updateServiceState(em, relatedService);
+        }
```

Comment:
Can this actually ever happen? I would assume that if >= maxAttempts is true here, the relatedService would already be in ERROR state. In that case it would be better to remove the whole _if_ here.

Reply:
It begs the question of why it's modifying the state of related services at all as surely if they failed themselves they would be caught be the code that precedes this for the current service. Assuming the good reason for this has been lost in time, there is no reason for the related services to the thrown into ERROR state while the parent service remains in WARNING state until MAX_ATTEMPTS is reached.

Reply:
Well, the general idea probably was that if a related service is in warning state and the job runs successfully on another service, it's obviously not a problem with the job, but the related service, so it would be thrown into error state regardless of the number of attempts. But this could lead to a related service being in error state after only one attempt, which really doesn't sound like a good idea. 
But I think your fix (considering the number of attempts before kicking a related service into error state) really makes the whole check of related services obsolete (which is maybe a good thing?)

Reply:
Confused. If the fix makes it work then the code around it can't be obsolete otherwise the fix wouldn't be there.

Reply:
OK I follow now (blame's lack of sleep)